sql/core/src/test/scala/org/apache/spark/sql/DatasetAggregatorSuite.scala

```diff
@@ -314,4 +343,14 @@ class DatasetAggregatorSuite extends QueryTest with SharedSQLContext {
     val ds3 = sql("SELECT 'Some String' AS b, 1279869254 AS a").as[AggData]
     assert(ds3.select(NameAgg.toColumn).schema.head.nullable === true)
   }
+
+  test("Aggregator on empty dataset") {
```

Comment:
I think we should have a unit test for this bug instead of this complex end-to-end test, e.g. we can create `Invoke` directly and call a method returning null.